sdk/storage/inc/datalake/datalake_file_system_client.hpp

```diff
@@ -0,0 +1,128 @@
+#pragma once
+
+#include "common/storage_credential.hpp"
+#include "common/storage_url_builder.hpp"
+#include "datalake_options.hpp"
+#include "http/pipeline.hpp"
+#include "protocol/datalake_rest_client.hpp"
+#include "datalake/datalake_service_client.hpp"
+
+#include <memory>
+#include <string>
+
+namespace Azure { namespace Storage { namespace DataLake {
+
+  class DataLakePathClient;
+
+  class DataLakeFileSystemClient {
+  public:
+    /**
+     * @brief Create from connection string
+     * @param connectionString Azure Storage connection string.
+     * @param fileSystemName The name of a file system.
+     * @param options Optional parameters used to initialize the client.
+     * @return DataLakeFileSystemClient
+     */
+    static DataLakeFileSystemClient CreateFromConnectionString(
+        const std::string& connectionString,
+        const std::string& fileSystemName,
+        const DataLakeFileSystemClientOptions& options = DataLakeFileSystemClientOptions());
+
+    /**
+     * @brief Shared key authentication client.
+     * @param fileSystemUri The URI of the file system this client's request targets.
+     * @param credential The shared key credential used to initialize the client.
+     * @param options Optional parameters used to initialize the client.
+     */
+    explicit DataLakeFileSystemClient(
+        const std::string& fileSystemUri,
+        std::shared_ptr<SharedKeyCredential> credential,
+        const DataLakeFileSystemClientOptions& options = DataLakeFileSystemClientOptions());
+
+    /**
+     * @brief Bearer token authentication client.
+     * @param fileSystemUri The URI of the file system this client's request targets.
+     * @param credential The token credential used to initialize the client.
+     * @param options Optional parameters used to initialize the client.
+     */
+    explicit DataLakeFileSystemClient(
+        const std::string& fileSystemUri,
+        std::shared_ptr<TokenCredential> credential,
+        const DataLakeFileSystemClientOptions& options = DataLakeFileSystemClientOptions());
+
+    /**
+     * @brief Anonymous/SAS/customized pipeline auth.
+     * @param fileSystemUri The URI of the file system this client's request targets.
+     * @param options Optional parameters used to initialize the client.
+     */
+    explicit DataLakeFileSystemClient(
+        const std::string& fileSystemUri,
+        const DataLakeFileSystemClientOptions& options = DataLakeFileSystemClientOptions());
+
+    /**
+     * @brief Create a DataLakePathClient from current DataLakeFileSystemClient
+     * @param path Path of the resource within the file system.
+     * @return DataLakePathClient
+     */
+     DataLakePathClient GetDataLakePathClient(const std::string& path) const;
+
+    /**
+     * @brief Creates the file system.
+     * @param options Optional parameters to create this file system.
+     * @return FileSystemCreateResponse
+    */
+    DataLakeRestClient::FileSystemCreateResponse Create(
```

Comment:
I feel these response types should be in Model namespace, since we don't have Model namespace for now. So I think they should be in `Azure::Storage::DataLake`. It's all about how your generator works.

Reply:
Changed and it is now in `Azure::Storage::DataLake`.